opensrp-reveal/src/main/java/org/smartregister/reveal/widget/RevealEditTextFactory.java

```diff
@@ -61,6 +66,19 @@ protected void attachLayout(String stepName, Context context, JsonFormFragment f
             hint.setSpan(new ForegroundColorSpan(Color.RED), hint.length() - 1, hint.length(), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
             editText.setHint(hint);
         }
+      
+        // truncate hint when typing
+        String shortenedHint = editText.getHint().toString().split("\\(")[0];
```

Comment:
Instead of this assumption maybe you could pass the shortened hint from the Json. So that its configurable